### PR TITLE
Helpchan: Add closed status

### PR DESCRIPTION
**Pull request by smichel17**
- Pins the current status, which has more information, instead of the asker's first message.
- When a channel is closed, updates the "ongoing" status embed to "closed" with a link to the next question, useful for skimming.
    - Possible because the status message is pinned now, so we don't have to fetch an arbitrary number of messages to find it.
- For newly-created help channels, make the topic refer to #how-to-get-help and the pinned status instead of "Ask your questions here!"
    - For existing help channels, the topic will need to be updated manually.

### CODE REVIEW

**Comment by smichel17** [comment]
For moving a channel from dormant to available, if there is no pinned status, it will send a new one.
From available to occupied, it will do nothing (it simply won't update the status). Same for occupied => closed.
A new dormant status will always be sent (and pinned), so once channels have rotated through, their state will be fine.

So, when this is merged, the two available help channels should be cycled. For everything else, the status of the pins will upgrade cleanly.

**Comment by smichel17** [comment]
Oh, right — also it updates the phrasing in the available and occupied status embeds. Now they are mostly-identical, so you don't need to re-read the bullets after claiming a channel, and overall improved, I think.

image
image

**Comment by smichel17** [comment]
(rebased on master)

**Comment by ronthecookie** [comment]
Is this ready for review?

**Comment by smichel17** [comment]
Yes

**Comment by smichel17** [comment]
A couple of the commits, which all update the status message text, really should be rebased/squashed, but when I tried it initially, they didn't re-order cleanly and I didn't want to mess up the other commits, which are relatively clean :confused:

**Review by robertt — APPROVED** [review]
I like it - do you think it would be possible to keep the first question pinned ongoing channels like before? I think it's useful to be able to quickly jump to the original question

**Comment by smichel17** [comment]
Technically there's no reason why not; we could pin both the status and the question. That said, the pinned status is always the message above the question, so it's still just as fast to jump to the first message. Pinning both *would* let you read the question in the pins without having to jump to it; I don't do that very often myself, since often the original question is split into a few messages, and it's harder for me to read the question in the pins. Happy to add it back if it's something other people do.

**Comment by smichel17** [comment]
Rebased & at least squashed the two commits that were next to each other

**Comment by robertt** [comment]
I think the original question being pinned was helpful - would it be possible for it to be pinned as well? Other then that, I'd be happy to merge this.

**Comment by smichel17** [comment]
Sure. I can add it tonight (won't be at at a computer until then). There's already a filter in place (it doesn't blindly assume that all pinned messages are status embeds), and all messages are unpinned when it is marked dormant, so it should just be a matter of re-adding the two lines to pin the question.

**Comment by smichel17** [comment]
s/tonight/tomorrow morning because I was exhausted/

Anyway, pinning the other message works fine, but at least with the current way it functions, it does mean that the question will always be above the status:

image

**Comment by smichel17** [comment]
Hold on, one more commit coming…

File: src/modules/helpchan.ts [hunk 1]
```diff
@@ -115,13 +115,7 @@ export class HelpChanModule extends Module {
...
 		return new MessageEmbed(this.OCCUPIED_EMBED_BASE)
 			.setDescription(occupiedMessage(asker))
-			.setFooter(
-				`Closes after ${
-					dormantChannelTimeout / 60 / 60 / 1000
-				} hours of inactivity or when ${
-					asker.displayName
-				} sends !close.`,
-			);
+			.setFooter(`Closes after ${dormantChannelTimeoutHours} hours of inactivity or when ${asker.username} sends !close.`);
```
Comment (smichel17) [inline src/modules/helpchan.ts]: I changed this to displayName on purpose, as it's easier to recognize someone by that

Comment (robertt) [reply src/modules/helpchan.ts]: Ahh sorry, I merged master into this branch and didn't notice that change.

**Comment by smichel17** [comment]
Okay, this is done/ready, needs fixing the build/lint but I need to head out in approximately 1 minute so I can't do it now, back later in the day.

**Comment by smichel17** [comment]
@robertt Okay, ready to go

File: src/modules/helpchan.ts (hunk 1 above)
Comment (smichel17) [reply src/modules/helpchan.ts]: Fixed in https://github.com/typescript-community/community-bot/pull/132/commits/e7b1135c4b49eba6e2a81f9d57a18f35499161ec

**Pull request merged**